### DISCUSSION TITLE
Overriding a @model_validator(mode="before") classmethod and calling parent.

### DISCUSSION THREAD

**Discussion by EsmeMaxwell** [pydantic/pydantic#8625] [Question]
Could I check that I'm not doing something stupid. In short, what I'd like to do is define a generic model that implements some model validation in the "before" mode and then inherit from that model. The descendent models need to implement their own additional "before" model validation but also call the model validation used in the parent.

I've quickly tested this and it 'works' but VSCode is suggesting there's something wrong with the typing.

```python
class TestAModel(BaseModel):

    a: int = 1

    @model_validator(mode="before")
    @classmethod
    def pre_process(cls, data: Any) -> Any:
        if isinstance(data, dict):
            if "a" in data.keys() and isinstance(data["a"], str):
                if data["a"] == "test":
                    data["a"] = 100
                else:
                    data["a"] = int(data["a"])

        return data


test_a_model = TestAModel(a="test")
pprint(test_a_model)
```

which gives what is sought:

```python
TestAModel(a=100)
```

Now inherit,

```python
class TestBModel(TestAModel):

    @classmethod
    def pre_process(cls, data: Any) -> Any:
        super(TestBModel, cls).pre_process(data)

        if isinstance(data, dict):
            if "a" in data.keys() and isinstance(data["a"], int):
                data["a"] = data["a"] + 12

        return data

test_b_model = TestBModel(a="test")
pprint(test_b_model)
```

which, again, gives what I wanted:

```python
TestBModel(a=112)
```

The issue is VSCode/Pylance is complaining with the method call `super(TestBModel, cls).pre_process(data)` with warning 'Object of type "PydanticDescriptorProxy[ModelValidatorDecoratorInfo]" is not callable'.

Can I just tell it to ignore the type issue or am I doing something fundamentally wrong?

**Comment by maparham** [comment]
Any updates on how to fix this?
